### PR TITLE
[WIP] Ndarray cumsum

### LINKED ISSUE #13001: Feature request: numpy.cumsum
MXNet equivalence of numpy.cumsum https://docs.scipy.org/doc/numpy-1.15.0/reference/generated/numpy.cumsum.html?highlight=cumsum#numpy.cumsum

### PR DESCRIPTION
**Pull request by ChaiBapchya**
## Description ##
Fixes #13001 
Numpy has cumsum - takes sum cumulatively. This PR is its NDArray API equivalent.

## Comments ##
- @eric-haibin-lin wanted your help on 2 things
1. implementing forward function
2. wanted to confirm api should look like this - 
```
mxnet.ndarray.cumsum(data=None, axis=_Null, keepdims=_Null, out=None, name=None, **kwargs)
```
or
```
mxnet.ndarray.cumsum(data=None, axis=_Null, out=None, name=None, **kwargs)
```

### CODE REVIEW

**Comment by ChaiBapchya** [comment]
@mxnet-label-bot add [pr-work-in-progress, Feature request]

**Comment by wkcn** [comment]
The in_grad is equal to cumsum(out_grad[::-1])

**Comment by vandanavk** [comment]
@reminisce for review

**Comment by ptrendx** [comment]
For GPU implementation please use CUB's scan: https://nvlabs.github.io/cub/structcub_1_1_device_scan.html

File: src/operator/tensor/cumulative_op.h [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+#include "../operator_common.h"
+#include "../elemwise_op_common.h"
+#include "../tensor/init_op.h"
+
+namespace mxnet {
+namespace op {
+
+struct CumsumParam : public dmlc::Parameter<CumsumParam> {
+  dmlc::optional<int> axis;
+  // bool keepdims; Do we need keepdims or not for cum_sum?
```
Comment (eric-haibin-lin) [inline src/operator/tensor/cumulative_op.h]: numpy does not have this option. Let's keep it consistent with numpy

**Comment by piyushghai** [comment]
@ChaiBapchya Can you address the review feedback on this PR ?

File: src/operator/tensor/cumulative_op.h (hunk 1 above)
Comment (ChaiBapchya) [reply src/operator/tensor/cumulative_op.h]: alright. there were other ops in mxnet which had `keepdims` hence was worried if we need to keep it consistent with that. But ya I'll remove it

**Comment by karan6181** [comment]
@ChaiBapchya Thank you for your contribution! Could you please look into CI failure and get this PR ready for review/merge? Thanks!

**Comment by ChaiBapchya** [comment]
Yes. Sorry for the delay.

**Comment by eric-haibin-lin** [comment]
@ChaiBapchya  Any update on this? This operator is used in the recent ON-LSTM ICLR paper

**Comment by ChaiBapchya** [comment]
Apologies for delay. Out on vacation. Will resume from July 4.

**Comment by neerajBaji** [comment]
Any updates on this?

**Comment by ChaiBapchya** [comment]
Handled by https://github.com/apache/incubator-mxnet/pull/15924
Closing in favor of that PR.

**Pull request closed without merging**